### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by gabi-250**
* Update README with more detailed build instructions.

    This adds the LLVM minimum version requirement and a few sentences
    about how to build the splitter.

* Add missing bang in crunch bang.

### CODE REVIEW

**Comment by ltratt** [comment]
Hmm, bors succeeded but nothing merged?! Let's try again and cross our fingers.

bors r+

**Comment by gabi-250** [comment]
bors try

**Comment by ltratt** [comment]
Ready for me to try again?

**Comment by gabi-250** [comment]
I don't think it's actually running anything -- the last lines of the build log are
```
Only purecap architectures, i.e. riscv64 and morello, are supported.
Container disconnected
program finished with exit code 0
elapsedTime=1.839265
```
which suggests we're not passing the arguments to the `.buildbot.sh` script. I'm trying to see what changed in our CI set up since we last ran tests in this repo.

**Comment by gabi-250** [comment]
bors try

**Comment by gabi-250** [comment]
This is now reporting the errors I had locally - we either need to change our `llvm-config` invocation (i.e. don't pass `--link-shared`), or ensure we build LLVM according to the instructions I added in the README.

File: .buildbot.sh [hunk 1]
```diff
@@ -2,124 +2,9 @@
...
-cd out-lua
-	make CC=$CC CFLAGS="-DLUA_USE_READLINE_DL -D__LP64__=1 $CONFIG_FLAGS" LDFLAGS="$CONFIG_FLAGS" LLVM_CONFIG=$CHERI/bin/llvm-config
-	# Copy the interpreter to the running qemu instance and execute the tests
-	PYTHONPATH=$PYTHONPATH SSHPORT=$SSHPORT SSH_OPTIONS=$SSH_OPTIONS python3 ../tests/run_cheri_tests.py "${args[@]}"
-	make clean
\ No newline at end of file
+echo "Running tests for 'morello-purecap' using QEMU..."
+./run_tests.sh morello-purecap
+echo "Running tests for 'riscv64-purecap' using QEMU."
+# TODO
```
Comment (gabi-250) [inline .buildbot.sh:10]: I'll add the risc-v test after I fix the other build issues.

**Comment by gabi-250** [comment]
> This is now reporting the errors I had locally - we either need to change our llvm-config invocation (i.e. don't pass --link-shared), or ensure we build LLVM according to the instructions I added in the README.

Actually, I'm not sure we _need_ `--link-shared` (I don't see why we couldn't just link the splitter statically). Let me remove that and update the build instructions accordingly.

**Comment by gabi-250** [comment]
bors try

**Comment by gabi-250** [comment]
It's failing for a different reason now: it can't find `libtinfo` (I am trying to figure out it's even trying to link against that).

**Comment by gabi-250** [comment]
> It's failing for a different reason now: it can't find `libtinfo` (I am trying to figure out it's even trying to link against that).

it's because of the `llvm-config` `--system-libs` flag:
```
–system-libs

    Print all the system libraries needed to link against the specified LLVM components, including any dependencies.
```

**Comment by gabi-250** [comment]
I _think_ we just need to update the docker image to install `libtinfo-dev`.

**Comment by 0152la** [comment]
Could you please confirm this locally manually (run a fresh container from the image, install the package, clone the repo, and run `.buildbot.sh`), and let me know? I'll update the Dockerfile if that's the only thing needed.

**Comment by gabi-250** [comment]
I can confirm installing `libtinfo-dev`, `wget`, and `golang` fixes the issue (I used the `capablevms/cheri_unittest` image).

**Comment by 0152la** [comment]
Thanks; I've added them to our buildbot Dockerfile. It'll be done automatically overnight, or I can trigger a rebuild now if you'd prefer.

**Comment by gabi-250** [comment]
FWIW the tests fail later down the line with
```
Traceback (most recent call last):
  File "/home/buildbot/llvm-function-split/out-lua/../tests/run_cheri_tests.py", line 26, in <module>
    test_scripts_dir = str(Path(importlib.util.find_spec("run_tests_common").origin).parent.absolute())
AttributeError: 'NoneType' object has no attribute 'origin'
```
but it sounds like it's just a matter of updating `run_cheri_tests.py`.

**Comment by gabi-250** [comment]
@0152la thank you! I don't mind waiting for the overnight build (I can debug the remaining issues locally)

**Comment by ltratt** [comment]
bors try

**Comment by gabi-250** [comment]
bors try

**Comment by gabi-250** [comment]
bors try

**Comment by gabi-250** [comment]
@ltratt The build is green now - can I squash?

**Comment by ltratt** [comment]
Please squash.

**Comment by gabi-250** [comment]
Squashed!

**Pull request merged**